src/deepsparse/transformers/haystack.py

```diff
@@ -0,0 +1,359 @@
+        """
+        return self.query_pipeline(texts).embeddings
+
+    def embed_documents(self, docs: List[Document]) -> List[numpy.ndarray]:
+        """
+        :param docs: list of document strings to embed
+        :return: list of embeddings for each document
+        """
+        passage_inputs = [doc.content for doc in docs]
+        return self.passage_pipeline(passage_inputs).embeddings
+
+    def _get_predictions(*args, **kwargs):
+        raise NotImplementedError()
+
+    def train(*args, **kwargs):
+        raise NotImplementedError()
+
+    def save(*args, **kwargs):
+        raise NotImplementedError()
+
+    def load(*args, **kwargs):
+        raise NotImplementedError()
+
+
+class _DeepSparseEmbeddingEncoder(_BaseEmbeddingEncoder):
+    """
+    Deepsparse implementation of Haystack EmbeddingEncoder
+
+    :param retriever: retriever that uses this encoder
+    :param pipeline_kwargs: extra arguments passed to EmbeddingExtractionPipeline
+    """
+
+    def __init__(self, retriever: DeepSparseEmbeddingRetriever, pipeline_kwargs):
+        self.embedding_pipeline = Pipeline.create(
+            "embedding_extraction",
+            model_path=retriever.model_path,
+            batch_size=retriever.batch_size,
+            sequence_length=retriever.max_seq_len,
+            emb_extraction_layer=retriever.emb_extraction_layer,
+            extraction_strategy=retriever.pooling_strategy,
+            show_progress_bar=retriever.progress_bar,
+            **pipeline_kwargs,
+        )
+
+        self.batch_size = retriever.batch_size
+        self.show_progress_bar = retriever.progress_bar
+        document_store = retriever.document_store
+        if document_store.similarity != "cosine":
+            _LOGGER.warning(
+                f"You are using document store embeddings with the "
+                f"{document_store.similarity} function. We recommend using "
+                "cosine instead. This can be set when initializing DocumentStore"
+            )
+
+    def embed(
+        self, texts: Union[List[List[str]], List[str], str]
+    ) -> List[numpy.ndarray]:
+        """
+        :param texts: list of strings to embed
+        :return: list of embeddings for each string
+        """
+        model_output = self.embedding_pipeline(texts)
+        embeddings = [numpy.array(embedding) for embedding in model_output.embeddings]
+        return embeddings
+
+    def embed_queries(self, texts: List[str]) -> List[numpy.ndarray]:
+        """
+        :param texts: list of query strings to embed
+        :return: list of embeddings for each query
+        """
+        return self.embed(texts)
+
+    def embed_documents(self, docs: List[Document]) -> List[numpy.ndarray]:
+        """
+        :param docs: list of document strings to embed
+        :return: list of embeddings for each document
+        """
+        passages = [d.content for d in docs]
+        return self.embed(passages)
+
+
+def print_pipeline_documents(
```

Comment:
is there a reason we would want to print rather than return as formatted string or just dict?

Reply:
This function is meant to extend `print_documents` which is provided by haystack to our output schema. A Haystack user who is integrating our components can easily swap out their `print_documents` calls with our `print_pipeline_documents`